src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -77,25 +84,44 @@ public SortClause buildAndValidate(ObjectNode sortNode) {
       return SortClause.mutable(SortExpression.collecetionVectorizeSort(vectorizeData));
     }
 
-    // Otherwise, use shared default processing
-    return super.buildAndValidate(sortNode);
+    // Otherwise, only "regular" sort expressions are left
+    // So let's validate the paths for the sort expressions
+    validateSortExpressionPaths(sortNode);
+
+    Iterator<Map.Entry<String, JsonNode>> fieldIter = sortNode.fields();
+    List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
+
+    while (fieldIter.hasNext()) {
+      Map.Entry<String, JsonNode> inner = fieldIter.next();
+      sortExpressions.add(buildSortExpression(inner.getKey(), inner.getValue()));
```

Comment:
first through was how is this diff to  table, but I get that for collections we do validate if the name exists.

Reply:
Right: and bigger part is that Table part will further diverge beyond column-name-existence checks so there is not much commonality.